### ISSUE TITLE
Cannot decode perl string json on perl <= 5.8.6

### ISSUE THREAD

**Issue by shoichikaji** [makamaka/JSON-PP#8]
Here is what I have done.

```
> wget http://search.cpan.org/CPAN/authors/id/M/MA/MAKAMAKA/JSON-PP-2.27203.tar.gz
> tar xzf JSON-PP-2.27203.tar.gz
> cd JSON-PP-2.27203

> plenv local 5.8.7
> perl -Ilib -MJSON::PP -e 'JSON::PP->new->decode(qq|{ "a":0, "b":"\x{c0}" }|)' && echo OK
OK

> plenv local 5.8.6
> perl -Ilib -MJSON::PP -e 'JSON::PP->new->decode(qq|{ "a":0, "b":"\x{c0}" }|)' && echo OK
, or } expected while parsing object/hash, at character offset 7 (before ", "b":"\x{c0}" }") at -e line 1
```

**Comment by miyagawa** [comment]
Note: should be fixed by #9. @makamaka or @shoichikaji can you close this?

**Comment by shoichikaji** [comment]
@miyagawa You're right, this issue was fixed by #9. I close this.
